gce/src/main/java/com/example/getstarted/basicactions/HealthServlet.java

```diff
@@ -0,0 +1,85 @@
+/* Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.getstarted.basicactions;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Servlet to respond to health checks.
+ * <p>
+ * <p>This servlet responds to all requests with the message "ok" and HTTP code
+ * 200. It is periodically run by the health checker to determine if the server
+ * is still responding to requests.
+ * <p>
+ * <p>A health check servlet like this is required when using the load balancer
+ * for Google Compute Engine, but App Engine flexible environment provides one
+ * for you if you do not supply one.
+ */
+@SuppressWarnings("serial")
+@WebServlet(
+    name = "health",
+    urlPatterns = {
```

Comment:
Why is the HealthServlet handling all 3 of the endpoints?

Reply:
It's copied from all the other HealthServlets in getting-started-java. What's the alternative?

Reply:
It looks like there are three separate endpoints: `/_ah/health`, `/_ah/start`, and `/_ah/stop`. I don't think there is any gain from having the health check responding on `/_ah/start` or `/_ah/stop`. (This my not even be necessary for Java, usually a ContextListener is used to create/destroy on servlet startup/shutdown)

Reply:
/_ah/health ... are AppEngine endpoints.  You don't need the _ah for anything else.

Reply:
Ah ok, I just removed that file entirely then.